clinica/pipelines/t1_linear/t1_linear_pipeline.py

```diff
@@ -0,0 +1,258 @@
+            ])
+
+        if (self.parameters.get('crop_image')):
+            self.connect([
+                (self.output_node, write_node, [('outfile_crop', '@outfile_crop')]),
+                ])
+
+    def build_core_nodes(self):
+        """Build and connect the core nodes of the pipeline.
+        """
+        import nipype.interfaces.utility as nutil
+        import nipype.pipeline.engine as npe
+        from clinica.utils.filemanip import get_subject_id
+        from clinica.utils.nipype import fix_join
+        from nipype.interfaces import ants
+        from .t1_linear_utils import (crop_nifti, container_from_filename, get_data_datasink)
+
+        image_id_node = npe.Node(
+                interface=nutil.Function(
+                    input_names=['bids_or_caps_file'],
+                    output_names=['image_id'],
+                    function=get_subject_id),
+                name='ImageID'
+                )
+
+        # The core (processing) nodes
+        # =====================================
+
+        # 1. N4biascorrection by ANTS. It uses nipype interface.
+        n4biascorrection = npe.Node(
+                name='n4biascorrection',
+                interface=ants.N4BiasFieldCorrection(
+                    dimension=3,
+                    save_bias=True,
+                    bspline_fitting_distance=600
+                    )
+                )
+
+        # 2. `RegistrationSynQuick` by *ANTS*. It uses nipype interface.
+        ants_registration_node = npe.Node(
+                name='antsRegistrationSynQuick',
+                interface=ants.RegistrationSynQuick()
+                )
+        ants_registration_node.inputs.fixed_image = self.ref_template
+        ants_registration_node.inputs.transform_type = 'a'
+        ants_registration_node.inputs.dimension = 3
+
+        # 3. Crop image (using nifti). It uses custom interface, from utils file
+
+        cropnifti = npe.Node(
+                name='cropnifti',
+                interface=nutil.Function(
+                    function=crop_nifti,
+                    input_names=['input_img', 'ref_crop'],
+                    output_names=['output_img', 'crop_template']
+                    )
+                )
+        cropnifti.inputs.ref_crop = self.ref_crop
+
+        # Other nodes
+        # =====================================
+        # Get substitutions to rename files
+        get_ids = npe.Node(
+                interface=nutil.Function(
+                    input_names=['image_id'],
+                    output_names=['image_id_out', 'subst_ls'],
+                    function=get_data_datasink),
+                name="GetIDs")
+        # Find container path from t1w filename
+        container_path = npe.Node(
+                nutil.Function(
+                    input_names=['bids_or_caps_filename'],
+                    output_names=['container'],
+                    function=container_from_filename),
+                name='ContainerPath')
```

Comment:
The get_ids/containter_path nodes are usually set in `build_output_node` method. See e.g. [`pet_volume_pipeline.py`](https://github.com/aramis-lab/clinica/blob/dev/clinica/pipelines/pet_volume/pet_volume_pipeline.py) to see its use

Reply:
Done!